### DISCUSSION TITLE
[Crud Service] header not supported by API Gateway with nginx

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#41] [Q&A] [answered]
Hi

In the `crud-service` it was recently added [an header](https://docs.mia-platform.eu/docs/runtime_suite/crud-service/overview_and_usage#crud-headers) to handle the value inside the `_q` query param as base64.

The parameter, named `json_query_params_encoding`, will be discarded if the `crud-service` is proxied by the `API Gateway` with `nginx`.

This is a [known issue](https://amalgjose.com/2021/07/12/how-to-fix-nginx-not-accepting-headers-with-underscore/) and can be solved by adding to `server-extension.conf ` the following line:

```
underscores_in_headers on;
```

Can we consider to change the name of the header?
In a multi-project environment, where several API Gateways are used, this change should be tracked to all the gateways.

**Comment by fredmaggiowski** [comment]
I agree, the use of `-` is the standard; an improvement on the service should be the best solution imho; can you take it in charge? @rstagi FYI

**Comment by rstagi** [comment, marked as answer]
Hi, the issue should be fixed with the new CRUD Service release, i.e. version`6.0.1`. Please update your service and you'll find a new header `json-query-params-encoding` which has the exact same behaviour of the old one.

Thank you for making us notice that :)
